### PR TITLE
Add types to InstructorAssessmentInstance page

### LINKED ISSUE #9246: Invalid instructor question links
Reported by @firasm

On [this page](https://ca.prairielearn.com/pl/course_instance/6665/instructor/assessment_instance/1156383) the "instructor view" link for the question is https://ca.prairielearn.com/pl/course_instance/6665/instructor/question// -- note the missing `question_id` between the last two slashes. All other links work fine.

Firas reports that both the AID (the assessment directory name) and the QID were renamed after the students completed their work. But that seems to be irrelevant because this actually seems broken on all instructor assessment instance pages.

## Code

The missing value is `instance_question.question_id` in the following line:
https://github.com/PrairieLearn/PrairieLearn/blob/8b4a523eb62877cb309df496127d6e879ec9a11a/apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ejs#L170

I think the problem was introduced by #9072 -- the `select_instance_questions` query in `instructorAssessmentInstance.sql` returns a `question_id` but it's missing from this type:
https://github.com/PrairieLearn/PrairieLearn/blob/8b4a523eb62877cb309df496127d6e879ec9a11a/apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js#L49-L53

There seem to be a lot of columns returned by that query that are missing from that type?

## Screenshot of the page above

image

### PR DESCRIPTION
**Pull request by mylesw27**
In PR #9072, we added zod-validation to our queries and add schemas to validate the data. However, as pointed out #9246, there were columns from the `select_instance_questions` query that were not included int he `InstanceQuestionRowSchema`. Thus, there were bugs and features missing from the page. This PR intends to add those missing types to the Zod schema and restore features to the page. Closes #9246.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js [hunk 1]
```diff
@@ -47,9 +47,24 @@ const DateDurationResultSchema = z.object({
 });
 
 const InstanceQuestionRowSchema = InstanceQuestionSchema.extend({
+  instructor_question_number: z.string().nullable(),
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js:50]: Is this actually nullable?

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js [hunk 2]
```diff
@@ -47,9 +47,24 @@ const DateDurationResultSchema = z.object({
...
+  manual_rubric_id: IdSchema.nullable(),
+  max_auto_points: z.number().nullable(),
+  max_manual_points: z.number().nullable(),
+  max_points: z.number().nullable(),
   modified_at: z.string(),
   qid: z.string().nullable(),
+  question_id: IdSchema,
   question_number: z.string().nullable(),
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js:58]: Is this actually nullable?

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js [hunk 3]
```diff
@@ -47,9 +47,24 @@ const DateDurationResultSchema = z.object({
...
+  question_title: z.string().nullable(),
+  row_order: z.number().nullable(),
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js:60]: Is this actually nullable?

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js [hunk 4]
```diff
@@ -47,9 +47,24 @@ const DateDurationResultSchema = z.object({
...
+  start_new_zone: z.boolean(),
+  zone_best_questions: z.number().nullable(),
+  zone_has_best_questions: z.boolean(),
+  zone_has_max_points: z.boolean(),
+  zone_id: IdSchema.nullable(),
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js:65]: Is this actually nullable?

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js (hunk 1 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js]: No. I now realize it is not nullable as the response is coming from the sproc `admin_assessment_question_number`, which always returns text. I have removed `.nullable()` in commit [7a88b09](https://github.com/PrairieLearn/PrairieLearn/pull/9250/commits/7a88b098dd62ef42b7196aa4cdafa015e6675a30).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js (hunk 2 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js]: No, this should never be null. Removed `.nullable()` in commit [7a88b09](https://github.com/PrairieLearn/PrairieLearn/pull/9250/commits/7a88b098dd62ef42b7196aa4cdafa015e6675a30).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js (hunk 3 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js]: No. I now realize this is coming from sproc `question_order` which should always return an integer here. Removed `.nullable()` in commit [7a88b09](https://github.com/PrairieLearn/PrairieLearn/pull/9250/commits/7a88b098dd62ef42b7196aa4cdafa015e6675a30).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js (hunk 4 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js]: No, this should never be nullable. Removed `.nullable()` in commit [9ccc082](https://github.com/PrairieLearn/PrairieLearn/pull/9250/commits/9ccc082ef35b88e51bdbf3789d2ef262dbd4fbfc).

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**